gamemodes/terrortown/gamemode/client/cl_transfer.lua

```diff
@@ -50,7 +50,7 @@ function CreateTransferMenu(parent)
 	for i = 1, #plys do
 		local ply = plys[i]
 
-		if ply ~= client and ply:IsActive() and (not ply:GetSubRoleData().unknownTeam or ply:IsRole(ROLE_DETECTIVE) and client:IsRole(ROLE_DETECTIVE)) and ply:IsInTeam(client) then
+		if ply ~= client and ply:IsActiveShopper() and (not ply:GetSubRoleData().unknownTeam or ply:GetBaseRole() == ROLE_DETECTIVE) then
```

Comment:
A Detective should be a shopper in every way, so the new check is useless. The goal of the old check have been to make sure just Detectives can send credits to each other AND all in all just teammates.

Additionally, preventing something on the server but not on the client is bad practice, because in that way the situation could happen that credits won't decrease / transfer even if it's a valid option for the client. So add the hook here as well or make it possible to transfer credits to everyone OR to transfer credits just to teammates. I don't think it's wrong to be able transferring credits to the Defective if thinking it's a Detective. Otherwise you could prove him.

Reply:
> A Detective should be a shopper in every way, so the new check is useless

So then should the check be just "ply ~= client and ply:IsActiveShopper()" ?

> So add the hook here as well.

Perhaps then the check would be something like: 
```lua
if ply ~= client and ply:IsActiveShopper() and hook.Run("TTT2CanTransferCredits", client, ply, 1) ~= false then
```
I can make that change and re-test.

> or make it possible to transfer credits to everyone

That's more or less what this code change means to do. The only catches are: the sender must have some idea as to what role the recipient has and that role must be a shopping role.

> I don't think it's wrong to be able transferring credits to the Defective if thinking it's a Detective. Otherwise you could prove him.

Agreed. The change here is a fix for the reverse: currently a Defective cannot transfer credits to a Detective because the Defective sees that they are on different teams. This change allows for cross-team credit transferring.

Reply:
I totally agree with the shared thingy.

But why is the detective always a shopper? Isn't it possible to remove the detective shop with the shopeditor?

Reply:
Hmmm... I thought about the topic again and regarding that amount of different roles, shouldn't it rather be possible transferring credits to EVERYONE? For a detective, it's still hard to transfer credits to a survivalist even if that role is proven (e.g. through gamestyle oder priest), because the only way to detect that role as proven is by killing the player and confirming (credits transfer at this point doesn't make sense anymore lmao)

Reply:
yeah, transfering it to everyone seems to be the easiest and cleanest solution. I like it

Reply:
By everyone does this include non-shopping roles? Can non-shoppers even transfer credits currently? I'm all for it, but I'm curious about what kind of changes this would require.

Reply:
Yes, I think this is the best solution. Otherwise you might be able to detect the innocent player. If you so desire to spend your credits someone that can not use them, it's your own fault.